engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManager.scala

```diff
@@ -72,11 +73,12 @@ class K8sDeploymentManager(modelData: ModelData, config: K8sDeploymentManagerCon
   override def deploy(processVersion: ProcessVersion, deploymentData: DeploymentData,
                       processDeploymentData: ProcessDeploymentData,
                       savepointPath: Option[String]): Future[Option[ExternalDeploymentId]] = {
+    val (scalingOptions, scenarioWithCorrectedParallelism) = determineScalingOptions(processDeploymentData)
     for {
-      configMap <- k8sUtils.createOrUpdate(k8s, configMapForData(processVersion, processDeploymentData))
+      configMap <- k8sUtils.createOrUpdate(k8s, configMapForData(processVersion, scenarioWithCorrectedParallelism))
```

Comment:
I wonder if it's the best way to achieve this. Maybe it would be better to pass a parameter to the image, saying how parallelism maps to threads?

Reply:
let's do this in next change